core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/NamespaceChangeStreamListener.java

```diff
@@ -101,16 +102,25 @@ public void start() {
    * Stops the background stream thread.
    */
   public void stop() {
+    if (runnerThread == null) {
+      return;
+    }
+
+    runnerThread.interrupt();
+
     nsLock.writeLock().lock();
     try {
       if (runnerThread == null) {
         return;
       }
 
       this.cancel();
+      this.close();
+
       while (runnerThread.isAlive()) {
         runnerThread.interrupt();
         try {
+          System.out.println("killing runner");
```

Comment:
[q] was this for debugging? delete if so. if this would be useful for users, make this an actual log item